src/clojure/nrepl/socket.clj

```diff
@@ -0,0 +1,239 @@
+(ns nrepl.socket
+  "Compatibility layer for java.io vs java.nio sockets to allow an
+  incremental transition to nio, since the JDK's filesystem sockets
+  don't support the java.io socket interface, and we can't use the
+  compatibility layer for bidirectional read and write:
+  https://bugs.openjdk.java.net/browse/JDK-4509080."
+  (:require
+   [clojure.java.io :as io]
+   [nrepl.misc :refer [log]])
+  (:import
+   (java.io BufferedInputStream BufferedOutputStream File)
+   (java.net InetSocketAddress ProtocolFamily ServerSocket Socket SocketAddress
+             StandardProtocolFamily URI)
+   (java.nio ByteBuffer)
+   (java.nio.file Path)
+   (java.nio.channels Channels ClosedChannelException NetworkChannel
+                      ServerSocketChannel SocketChannel)))
+
+(def orig-warn-on-reflection *warn-on-reflection*)
```

Comment:
Unless I misunderstand, the problem is that it'll break the tests (CI) right now, since it looks like reflection warnings were recently made errors.  That's why I've been reworking the pr to remove all reflections.

Reply:
They're not really part of a hard CI check right now. Still please do try getting rid of the reflection warnings, I think it's possible.

Reply:
Hmm, I thought they were now.  I must have conflated the cause of some earlier failures.